clang/utils/TableGen/MveEmitter.cpp

```diff
@@ -1415,7 +1415,10 @@ ACLEIntrinsic::ACLEIntrinsic(EmitterBase &ME, const Record *R,
         } else if (Bounds->getName() == "IB_LaneIndex") {
           IA.boundsType = ImmediateArg::BoundsType::ExplicitRange;
           IA.i1 = 0;
-          IA.i2 = 128 / Param->sizeInBits() - 1;
+          unsigned sizeInBits = Param->sizeInBits();
+          if (sizeInBits == 0)
+            PrintFatalError("Division by zero: Param->sizeInBits() is zero.");
```

Comment:
This doesn't seem like a useful diagnostic to someone working on TableGen code. The issue is that the param should never have a zero size to begin with (because that logically doesn't make sense, right?).

Reply:
Param is just some `Type` here, which can be `void`, returning 0 for its size.

Reply:
mm
https://github.com/llvm/llvm-project/blob/97298853b4de70dbce9c0a140ac38e3ac179e02e/clang/utils/TableGen/MveEmitter.cpp#L102

Reply:
Oh! Good catch (and interesting to know)!

Regardless, this diagnostic is saying what the code is doing wrong, not what the user of tablegen is doing wrong. The diagnostic should be written such that it helps the developer figure out how to fix their code.